tests/components/airthings_ble/test_config_flow.py

```diff
@@ -0,0 +1,176 @@
+"""Test the Airthings BLE config flow."""
+from unittest.mock import patch
+
+from airthings_ble import AirthingsDevice
+from bleak import BleakError
+import pytest
+
+from homeassistant.components.airthings_ble.const import DOMAIN
+from homeassistant.config_entries import SOURCE_BLUETOOTH, SOURCE_USER
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from . import (
+    UNKNOWN_SERVICE_INFO,
+    WAVE_DEVICE_INFO,
+    WAVE_SERVICE_INFO,
+    patch_airthings_ble,
+    patch_async_ble_device_from_address,
+    patch_async_setup_entry,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_bluetooth_discovery(hass: HomeAssistant):
+    """Test discovery via bluetooth with a valid device."""
+    with patch_async_ble_device_from_address(WAVE_SERVICE_INFO):
+        with patch_airthings_ble(
+            AirthingsDevice(name="Airthings Wave+", identifier="CCCCCC")
+        ):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_BLUETOOTH},
+                data=WAVE_SERVICE_INFO,
+            )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "bluetooth_confirm"
+    assert result["description_placeholders"] == {"name": "Airthings Wave+ CCCCCC"}
+
+    with patch_async_setup_entry():
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={"not": "empty"}
+        )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Airthings Wave+ CCCCCC"
+    assert result["result"].unique_id == "cc:cc:cc:cc:cc:cc"
+
+
+async def test_bluetooth_discovery_no_BLEDevice(hass: HomeAssistant):
+    """Test discovery via bluetooth but there's no BLEDevice."""
+    with patch_async_ble_device_from_address(None):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_BLUETOOTH},
+            data=WAVE_SERVICE_INFO,
+        )
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+async def test_bluetooth_discovery_airthings_ble_update_failed(
+    hass: HomeAssistant,
+):
+    """Test discovery via bluetooth but there's an exception from airthings-ble."""
+    with pytest.raises(UpdateFailed):
+        with patch_async_ble_device_from_address(WAVE_SERVICE_INFO):
+            with patch_airthings_ble(side_effect=UpdateFailed("fail!")):
+                await hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_BLUETOOTH},
+                    data=WAVE_SERVICE_INFO,
+                )
```

Comment:
Looks like this test needs to be adjusted. We should test the result of the config flow.   These aren't expected to raise exceptions since all the responses are sent over the websocket api which can't handle exceptions.

Reply:
Yeah, forgot to change the test. Also added a general exception handling now as well. Saw that there's an "unknown"  reason too.